### PR TITLE
AEA-981 - Add flags to switch registry mode for fetch, publish and upgrade

### LINKED ISSUE #1816: Add support for mixed registry mode for aea fetch and aea publish
**Is your feature request related to a problem? Please describe.**
Currently, CLI uses by default remote registry, and by flag the local one. When working locally, we sometimes want to fetch, publish or upgrade a local agent project which has some remote dependencies and some local ones.

**Describe the solution you'd like**
Make mixed registry mode available via flag `--mixed`.

Fetch: agent project is fetched from local, we first look at local registry for packages, if not present, we check remote, if still not present, we fail.
Publish: agent project is published to local, we first check if dependencies are on local registry, then if they are on remote registry, if still not present, we fail.
Upgrade: we first upgrade based on local packages, then based on remote packages

This will help during local development.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Add flags to switch registry mode for fetch, publish and upgrade.
The flags are `--local`, `--remote`, and `--mixed`. They are mutually exclusive.

## Fixes

fixes #1816

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
n/a

### CODE REVIEW

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Comment by marcofavorito** [comment]
TODO: need to restore original help message for flags.

File: aea/cli/fetch.py [hunk 1]
```diff
@@ -120,30 +125,55 @@ def fetch_agent_locally(
...
+        add_item(ctx, item_type, item_id)
+    except click.ClickException as e:
+        if not ctx.config.get("is_mixed", False):
+            _raise(item_type, item_id, e)
+        ctx.set_config("is_local", False)
+        try:
+            add_item(ctx, item_type, item_id)
+        except click.ClickException as e:
+            _raise(item_type, item_id, e)
+        ctx.set_config("is_local", True)
```
Comment (marcofavorito) [inline aea/cli/fetch.py:179]: first we try locally, then we try remotely (`is_local` set to `False`) but only if `is_mixed` is True.

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**